weed/storage/volume_vacuum.go

```diff
@@ -261,6 +270,12 @@ func (v *Volume) makeupDiff(newDatFileName, newIdxFileName, oldDatFileName, oldI
 	if idx, err = os.OpenFile(newIdxFileName, os.O_RDWR, 0644); err != nil {
 		return fmt.Errorf("open idx file %s failed: %v", newIdxFileName, err)
 	}
+
+	stat, err := idx.Stat()
```

Comment:
move this section below `defer idx.Close()`

Reply:
Done